pandas/core/arrays/arrow/dtype.py

```diff
@@ -321,3 +326,27 @@ def __from_arrow__(self, array: pa.Array | pa.ChunkedArray):
         array_class = self.construct_array_type()
         arr = array.cast(self.pyarrow_dtype, safe=True)
         return array_class(arr)
+
+    def _maybe_promote(self, item: Any) -> tuple[DtypeObj, Any]:
+        if isinstance(item, pa.Scalar):
+            if not item.is_valid:
```

Comment:
Shouldn't NA always be able to be inserted into ArrowExtensionArray?

Reply:
im not sure.  pyarrow nulls are typed, so we could plausibly want to disallow e.g. `<pyarrow.TimestampScalar: None>` in a pyarrow integer dtype